src/workerd/server/server.c++

```diff
@@ -2494,6 +2497,94 @@ kj::Own<Server::Service> Server::makeWorker(kj::StringPtr name, config::Worker::
     isolateRegistrar->registerIsolate(name, isolate.get());
   }
 
+  if (conf.hasModuleFallback()) {
+    KJ_REQUIRE(experimental,
+               "The module fallback service is an experimental feature. "
+               "You must run workerd with `--experimental` to use this feature.");
+    // If the config has the moduleFallback option, then we are going to set up the ability
+    // to load certain modules from a fallback service. This is generally intended for local
+    // dev/testing purposes only.
+    auto& apiIsolate = isolate->getApiIsolate();
+    apiIsolate.setModuleFallbackCallback(
+        [address=kj::str(conf.getModuleFallback()), featureFlags=apiIsolate.getFeatureFlags()]
+        (jsg::Lock& js,
+         kj::StringPtr specifier,
+         jsg::CompilationObserver& observer,
+         jsg::ModuleRegistry::ResolveMethod method) mutable
+            -> kj::Maybe<jsg::ModuleRegistry::ModuleInfo> {
+      kj::Maybe<kj::String> jsonPayload;
+      {
+        // Module loading in workerd is expected to be synchronous but we need to perform
+        // an async HTTP request to the fallback service. To accomplish that we wrap the
+        // actual request in a kj::Thread, perform the GET, and drop the thread immediately
+        // so that the destructor joins the current thread (blocking it). The thread will
+        // either set the jsonPayload variable or not.
+        kj::Thread loaderThread(
+          [specifier,address=address.asPtr(),&jsonPayload, method]() mutable {
+          try {
+            const auto toStr = [](jsg::ModuleRegistry::ResolveMethod method) {
+              switch (method) {
+                case jsg::ModuleRegistry::ResolveMethod::IMPORT: return "import"_kjc;
+                case jsg::ModuleRegistry::ResolveMethod::REQUIRE: return "require"_kjc;
+              }
+              KJ_UNREACHABLE;
+            };
+
+            kj::AsyncIoContext io = kj::setupAsyncIo();
+
+            kj::HttpHeaderTable::Builder builder;
+            kj::HttpHeaderId kMethod = builder.add("resolve-method");
```

Comment:
Nitpicks:

* In KJ code we normally capitalize this like `Resolve-Method`, though it doesn't really matter.
* Since this isn't a standard header it should probably have an `X-` prefix?

Reply:
@mrbbot ... just a heads up, I went ahead and renamed the header to `x-resolve-method`.